homework/index.js

```diff
@@ -30,18 +32,82 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
+  function handleError(error) {
+    const root = document.getElementById('root');
+    removeChildren(root);
+    createAndAppend('div', root, { text: error.message, class: 'alert-error' });
+  }
+
+  function pageConstructor(data) {
+    data.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
+    const headerDiv = createAndAppend('div', root, { class: 'headerDiv' });
+    createAndAppend('p', headerDiv, { text: 'HYF Repositories' });
+    const headerSelect = createAndAppend('select', headerDiv, { class: 'headerSelect' });
+    data.map((elem, index) =>
+      createAndAppend('option', headerSelect, { text: elem.name, value: index }),
+    );
+    const container = createAndAppend('div', root, { class: 'container' });
+    const repoInfoDiv = createAndAppend('div', container, { class: 'repoInfoDiv' });
+    const ul = createAndAppend('ul', repoInfoDiv);
+    const repository = createAndAppend('li', ul, { text: `Repository :` });
+    createAndAppend('a', repository, {
+      text: `${data[0].name}`,
+      href: data[0].html_url,
+      target: '_blank',
+      id: 'repository',
+    });
+    createAndAppend('li', ul, { text: `Description :${data[0].description}`, id: 'description' });
+    createAndAppend('li', ul, { text: `Forks :${data[0].forks}`, id: 'forks' });
+    createAndAppend('li', ul, { text: `Update :${data[0].updated_at}`, id: 'update' });
```

Comment:
The _updated_at_ date in repo info is in ISO 8601 format, see https://developer.github.com/v3/#schema
Have a look at [JavaScript Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) to find a method that shows the date in readable format.